pycozmo/behavior.py

```diff
@@ -44,21 +177,75 @@ def __init__(self,
         self.id = str(behavior_id)
         self.needs_action_id = str(needs_action_id)
         self.display_name_key = str(display_name_key)
-        self.params = dict(params)
+        self.params = params
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(behavior_class=data['behaviorClass'],
+                   behavior_id=data['behaviorID'],
+                   needs_action_id=data.get('needsActionID', ''),
+                   display_name_key=data.get('displayNameKey', ''),
+                   params=BehaviorParameters.from_json(data.get('params')) if 'params' in data else None)
 
 
-def load_behaviors() -> Dict[str, Behavior]:
-    # TODO: Load cozmo_resources/config/engine/behaviorSystem/behaviors/*/*.json
+class TestBehavior:
+    def __init__(self,
+                 behavior_class: str,
+                 behavior_id: str,
+                 loop_forever: Optional[bool] = False,
+                 test_gap: Optional[float] = 0.0,
+                 runs_per_test: Optional[int] = 1) -> None:
+        self.behavior_class = str(behavior_class)
+        self.id = str(behavior_id)
+        self.loop_forever = loop_forever
+        self.test_gap = test_gap
+        self.runs_per_test = runs_per_test
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(behavior_class=data['behaviorClass'],
+                   behavior_id=data['behaviorID'],
+                   loop_forever=data.get('loopForever', False),
+                   test_gap=data.get('gapBetweenTests_s', 0.0),
+                   runs_per_test=data.get('nRunsPerTest', 1))
+
+
+def load_behaviors(resource_dir: str) -> Dict[str, Behavior]:
+    behavior_files = get_json_files(resource_dir,
+                                    ['/cozmo_resources/config/engine/behaviorSystem/behaviors/'])
+    behaviors = {}
+
+    for filename in behavior_files:
+        json_data = load_json_file(filename)
+        if 'Test' not in json_data['behaviorID']:
+            behaviors[json_data['behaviorID']] = Behavior.from_json(json_data)
+    return behaviors
+
+
+def load_test_behaviors(resource_dir: str) -> Dict[str, Behavior]:
+    behavior_files = get_json_files(resource_dir,
+                                    ['/cozmo_resources/config/engine/behaviorSystem/behaviors/'])
     behaviors = {}
+
+    for filename in behavior_files:
+        json_data = load_json_file(filename)
+        if 'Test' in json_data['behaviorID']:
+            behaviors[json_data['behaviorID']] = TestBehavior.from_json(json_data)
     return behaviors
 
 
-def load_reaction_trigger_behavior_map() -> Dict[str, ReactionTrigger]:
-    # TODO: Load cozmo_resources/config/engine/behaviorSystem/reactionTrigger_behavior_map.json
+def load_reaction_trigger_behavior_map(resource_dir: str) -> Dict[str, ReactionTrigger]:
     reaction_trigger_behavior_map = {}
+    filename = resource_dir + \
+        '/cozmo_resources/config/engine/behaviorSystem/reactionTrigger_behavior_map.json'
```

Comment:
PyCozmo targets to be platform independent so we should use `os.path.join()` or `pathlib` in cases like this.